it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,403 @@
+import org.corfudb.universe.logging.LoggingParams;
+import org.corfudb.universe.node.Node;
+import org.corfudb.universe.node.NodeException;
+import org.corfudb.universe.node.server.AbstractCorfuServer;
+import org.corfudb.universe.node.server.CorfuServer;
+import org.corfudb.universe.node.server.CorfuServerParams;
+import org.corfudb.universe.node.stress.Stress;
+import org.corfudb.universe.universe.UniverseParams;
+import org.corfudb.universe.util.IpTablesUtil;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Implements a docker instance representing a {@link CorfuServer}.
+ */
+@Slf4j
+public class DockerCorfuServer extends AbstractCorfuServer<CorfuServerParams, UniverseParams> {
+    private static final String IMAGE_NAME = "corfu-server:" + getAppVersion();
+    private static final String ALL_NETWORK_INTERFACES = "0.0.0.0";
+
+    @NonNull
+    private final DockerClient docker;
+    @NonNull
+    private final LoggingParams loggingParams;
+    private final AtomicReference<String> ipAddress = new AtomicReference<>();
+    private final AtomicBoolean destroyed = new AtomicBoolean();
+
+    @Builder
+    public DockerCorfuServer(DockerClient docker, CorfuServerParams params, UniverseParams universeParams,
+                             LoggingParams loggingParams) {
+        super(params, universeParams);
+        this.docker = docker;
+        this.loggingParams = loggingParams;
+    }
+
+    /**
+     * Deploys a Corfu server / docker container
+     */
+    @Override
+    public DockerCorfuServer deploy() {
+        log.info("Deploying the Corfu server. Docker container: {}", params.getName());
+
+        deployContainer();
+
+        return this;
+    }
+
+    /**
+     * This method attempts to gracefully stop the Corfu server. After timeout, it will kill the Corfu server.
+     *
+     * @param timeout a duration after which the stop will kill the server
+     * @throws NodeException this exception will be thrown if the server cannot be stopped.
+     */
+    @Override
+    public void stop(Duration timeout) {
+        log.info("Stopping the Corfu server. Docker container: {}", params.getName());
+
+        try {
+            ContainerInfo container = docker.inspectContainer(params.getName());
+            if (!container.state().running() && !container.state().paused()) {
+                log.warn("The container `{}` is already stopped", container.name());
+                return;
+            }
+            docker.stopContainer(params.getName(), (int) timeout.getSeconds());
+        } catch (DockerException | InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)